x/gamm/pool-models/balancer/amm_test.go

```diff
@@ -7,11 +7,15 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/stretchr/testify/require"
 
-	"github.com/osmosis-labs/osmosis/v7/osmoutils"
 	"github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/balancer"
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/internal/test_helpers"
 	"github.com/osmosis-labs/osmosis/v7/x/gamm/types"
 )
 
+type BalancerTestSuite struct {
```

Comment:
Should we instead merge `KeeperTestSuite` and `BalancerTestSuite` but keep the `BalancerTestSuite` name? I think all tests on a common abstraction layer should belong to the same suite and be named with the respect to the abstraction layer it is testing

Reply:
wdym by "merging" `KeeperTestSuite` and `BalancerTestSuite`?

Reply:
Here's the `BalancerTestSuite`: https://github.com/osmosis-labs/osmosis/blob/9488499f200642666ee3fe5d102ab197ebbcafde/x/gamm/pool-models/balancer/amm_test.go#L15

In the same file we are using `KeeperTestSuite`: https://github.com/osmosis-labs/osmosis/blob/9488499f200642666ee3fe5d102ab197ebbcafde/x/gamm/pool-models/balancer/amm_test.go#L21

Which is defined in `suite_test.go`: https://github.com/osmosis-labs/osmosis/blob/ce13399bf5271ab5827e56f9093149cb6b84feb0/x/gamm/pool-models/balancer/suite_test.go#L12

I see two potential problems here:
1. There are 2 suites on the same layer of abstraction (`balancer`)
   a. `KeeperTestSuite`
   b. `BalancerTestSuite`
2.  `KeeperTestSuite` name doesn't make sense because this is not testing the keeper layer

So I suggest removing `KeeperTestSuite` and porting all its tests to the `BalancerTestSuite` so that we have one suite in the `balancer` package

Reply:
@p0mvn that makes sense to me as well

Reply:
Oh yeah definitely makes sense! sgtm

Reply:
The original reason I left two different Suites was because of the two following reasons.
1. Having ` BalancerTestSuite` embed both `apptesting.KeeperTestHelper`, test_helpers.CfmmCommonTestSuite` causes error since we can't embed two same fields in one struct, and `apptesting.KeeperTestHelper` and `test_helpers.CfmmCommonTestSuite` both embed suite.Suite as default. 

We have a couple options as a get around for this, this one I can think of right now is removing suite.Suite embedding from `KeeperTestHelper`and setting it as a field like this 
```
type KeeperTestHelper struct {
	ApptestingSuite suite.Suite
	App             *app.OsmosisApp
	Ctx             sdk.Context
	QueryHelper     *baseapp.QueryServiceTestHelper
	TestAccs        []sdk.AccAddress
}
```

What do you think?

Reply:
cool, leaving as it is!